common-libs/user-management/src/CognitoIdentityService.ts

```diff
@@ -198,24 +213,9 @@ export class CognitoIdentityService {
       const cognitoTokens = this._normalizeTokensFromCognitoAuthenticationResult(result.AuthenticationResult)
       return { result: CompleteLoginPasswordlessResult.Success, cognitoTokens }
     } catch (error) {
-      // NOTE: IF below is a fix: when user first time enters wrong OTP, Cognito
-      //       returns new session, for which correct OTP wont work, so user
-      //       will be stuck until server restart and user enters OTP correctly
-      //       from the 1st attempt
-      if (error?.code === 'COR-5' && error?.inputParams?.token) {
-        const { token } = error.inputParams
-
-        try {
-          const parsedToken = JSON.parse(token)
-
-          tempSession[USERNAME] = parsedToken?.logInToken?.Session
-
-        } catch (parseError) {
-          // throw new SdkError('COR-5', {}, parseError)
-          // TODO: How to pass original error?
-          return { result: CompleteLoginPasswordlessResult.ConfirmationCodeWrong }
-        }
-      }
+      // NOTE: not deleted sessions after any errors will block user session
+      delete tempSession[hashedTokenSession]
```

Comment:
This code will only have an effect in a scenario like this:
1. User initiated passwordless login; cognito returned session A;
2. User entered a wrong OTP; we sent request to cognito with wrong OTP and session A; cognito returned session B;
3. User entered some OTP; we sent request to cognito with that new OTP and session B; there was an error from cognito (e.g. timeout or something).

Does this mean that when user retries the request (or enters some third OTP), we should use session A instead of B? Why?

Reply:
this code will have effect on any error i already add integration test cases 
so cases 
1) user entered OTP wrongly 3 times 
2) user entered OTP once and that OTP expired 
3) any failure will block a session in a map forever

Reply:
So if cognito was not available, an user should attempt to log in again instead of retrying confirmation with their existing session.

Reply:
we just do a clean up and avoiding a dead record there

Reply:
We probably need to communicate this fact to clients somehow?

> So if cognito was not available, an user should attempt to log in again instead of retrying confirmation with their existing session.

Currently it might be expected that if an attempt to confirm login returned some unknown error we can retry to confirm login again.
As I understand, with the current implementation we won't be able to confirm login anymore (we'll keep getting unknown errors probably?), so we need to restart the login flow.
Is my understanding correct?